Code changes to expose boolean retained flag from paho in order to al…

sky-sharma opened the pull request:
…low topic messages to be retained.

1. In definition of publish method that takes a String message param,
added "false" (default value of retained) to list of params in call to
underlying "publish".
2. In definition of publish that takes a Byte array payload param, added
"boolean retained" to list of params (int qos was already there).
3. Added a version of the publish method that takes a String message
param and a boolean retained param.

sky-sharma committed:
Code changes to expose boolean retained flag from paho in order to allow topic messages to be retained.

1. In definition of publish method that takes a String message param,
added "false" (default value of retained) to list of params in call to
underlying "publish".
2. In definition of publish that takes a Byte array payload param, added
"boolean retained" to list of params (int qos was already there).
3. Added a version of the publish method that takes a String message
param and a boolean retained param.

danielrs commented on src/main/java/com/clearblade/java/api/MqttClient.java:
```diff
@@ -208,17 +208,28 @@ public void publish(String topic, String message) throws ClearBladeException {
 		}
 	}
 
+	/**
+	 * Publishes the given message to the given topic.
+	 * @param topic topic to publish to
+	 * @param message message to publish
+	 * @param retained flag to set whether topic message is retained
+	 * @throws ClearBladeException if publish fails
+	 */
+	public void publish(String topic, String message, boolean retained) throws ClearBladeException {
+		publish(topic, message.getBytes(), defaultQualityOfService, retained);
+	}
+
```
Can we move this function so that it appears before our final publish function (currently line 211)

sky-sharma replied:
Done. Re-committed, but also committed change to UserAuthExample.java. See my extended description. I needed the "sleep" at the end to make sure a retained message was deleted (if I uncommented the line that deletes the retained message).

sky-sharma committed:
Moved new publish below final publish as per Daniel's suggestion in PR. Other changes (see extended descr.)

In MqttClient.java:
1. As per Daniel's suggestion moved new publish below final publish.
2. Added comment re retained flag in comments for final publish.

In UserAuthExample.java
1. In final for loop added code (some commented out) to give examples of
how to publish with or without retained flag.
2. Added commented out code that showed how to delete retained message.
Found that needed to use the TimeUnit.SECONDS.sleep in order for the
retained message to get deleted.
